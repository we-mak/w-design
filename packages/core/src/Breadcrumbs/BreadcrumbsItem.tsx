import React, { ReactNode } from "react";
import { BreadcrumbsItemWrapper, Seperator } from "./Styled";
import { GlobProps } from "../common/props";

export interface BreadcrumbsItemProps extends GlobProps {
  isCurrentPage?: boolean;
  children?: ReactNode;
}

export const BreadcrumbsItem = ({ isCurrentPage, children }: BreadcrumbsItemProps) => {
  return (
    <BreadcrumbsItemWrapper>
      {children}
      {!isCurrentPage && <Seperator role="presentation">/</Seperator>}
    </BreadcrumbsItemWrapper>
  );
};
