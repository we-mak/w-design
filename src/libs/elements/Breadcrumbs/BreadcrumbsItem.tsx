import React from "react";
import { BreadcrumbsItemWrapper, Seperator } from "./Styled";
import { BreadcrumbsItemProps } from "./types";

export const BreadcrumbsItem = ({ isCurrentPage, children }: BreadcrumbsItemProps) => {
  return (
    <BreadcrumbsItemWrapper>
      {children}
      {!isCurrentPage && <Seperator role="presentation">/</Seperator>}
    </BreadcrumbsItemWrapper>
  );
};
