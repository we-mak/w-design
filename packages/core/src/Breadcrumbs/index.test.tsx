import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import { BreadcrumbsItem } from "./BreadcrumbsItem";
import { Provider, Breadcrumbs } from "..";

afterEach(cleanup);

describe("<Breadcrumbs/>", () => {
  it("should render breadcrumbs correctly", () => {
    const { container } = render(
      <Provider>
        <Breadcrumbs>
          <BreadcrumbsItem>1</BreadcrumbsItem>
          <BreadcrumbsItem>2</BreadcrumbsItem>
          <BreadcrumbsItem isCurrentPage>Item 3</BreadcrumbsItem>
        </Breadcrumbs>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
