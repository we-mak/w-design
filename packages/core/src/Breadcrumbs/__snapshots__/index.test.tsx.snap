// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs/> should render breadcrumbs correctly 1`] = `
.c0 {
  color: #566E82;
  font-size: 0.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  line-height: 1.2;
  height: 1.2rem;
  box-sizing: border-box;
  max-width: 100%;
  padding: 0px;
}

.c1 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #008CC0;
}

.c2 {
  color: #566E82;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  text-align: center;
  padding: 0px 0.2rem;
}

<div>
  <nav
    aria-label="breadcrumb"
    class="c0"
  >
    <div
      class="c1"
    >
      1
      <div
        class="c2"
        role="presentation"
      >
        /
      </div>
    </div>
    <div
      class="c1"
    >
      2
      <div
        class="c2"
        role="presentation"
      >
        /
      </div>
    </div>
    <div
      class="c1"
    >
      Item 3
    </div>
  </nav>
</div>
`;
